hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveStylePartitionValueExtractor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Extractor for Hive Style Partitioned tables, when the parition folders are key value pairs.
+ *
+ * <p>This implementation extracts the partition value of yyyy-mm-dd from the path of type datestr=yyyy-mm-dd.
+ */
+public class HiveStylePartitionValueExtractor implements PartitionValueExtractor {
+  private static final long serialVersionUID = 1L;
+
+  @Override
+  public List<String> extractPartitionValuesInPath(String partitionPath) {
+    // partition path is expected to be in this format partition_key=partition_value.
+    String[] splits = partitionPath.split("=");
```

Comment:
may be I am being very nit picky. but is there a chance that partitionpath has "=" in its field name? I know it does not make sense. anyways. 
partitionpath field name = "datestr" in your example.
can it be "partition=path" ? 
If yes, Collections.singletonList(splits[1]) at line 40 might break right? 
@n3nash : do we make any such assumptions in hudi's code base wrt partition path's field name?

Reply:
Good question!
I have recreated the same scenario with `=` on both partition key and partition value, but both Hive and Spark encodes the `=` to `%3D` on both the partition keys and values.

Look at this folder:
`
/tmp.db/ds_test/date%3Dstr=2021%3D04%3D09
/tmp.db/ds_test/date%3Dstr=2021-04-08
`
Based on this test, we can see that the code won't break, let me know if I have missed anything.

[Here](https://github.com/apache/hudi/blob/3a926aacf6552fc06005db4a7880a233db904330/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java#L143) is the Hudi code to create hive style partitions.

Reply:
awesome, sounds good.